packages/dapi/lib/mediators/app-mediator.js

```diff
@@ -0,0 +1,10 @@
+const { EventEmitter } = require('events');
+
+class AppMediator extends EventEmitter {
```

Comment:
We had a discussion about the naming with the @markin-io and had some uncertains about that

Reply:
Thank you for your work! And especially for comprehensive test coverage for caching functionality.

I reviewed the whole PR, and came up with the next idea:

Instead of having `AppMediator`, and manually managing `cache` [here](https://github.com/dashevo/platform/blob/68d64dbb22157e2f8b689461f6a5783bf2d1baaa/packages/dapi/lib/grpcServer/handlers/blockheaders-stream/subscribeToNewBlockHeaders.js#L64) and [here](https://github.com/dashevo/platform/blob/68d64dbb22157e2f8b689461f6a5783bf2d1baaa/packages/dapi/lib/grpcServer/handlers/blockheaders-stream/getHistoricalBlockHeadersIteratorFactory.js#L45), I'd suggest having the next class:

<img width="193" alt="Screenshot 2022-02-10 at 10 44 40" src="https://user-images.githubusercontent.com/3499359/153333330-bf8e1d91-85d0-4acf-9a88-4a66a264ee46.png">

The main purpose of this class would be encapsulating cache, core RPC calls, and ZMQ events.

Benefits would be:
- Abstracting API calls to ZMQ and Core RPC API
- Encapsulating complex [cache manipulation](https://github.com/dashevo/platform/blob/68d64dbb22157e2f8b689461f6a5783bf2d1baaa/packages/dapi/lib/grpcServer/handlers/blockheaders-stream/getHistoricalBlockHeadersIteratorFactory.js#L51) logic
- More granular testing. E.g. instead of testing whole `subscribeToBlockHeadersWithChainLocks` we would just need to cover `ChainDataHandler` with tests
- Better readability of the code

@shuplenkov @jawid-h can you guys please validate this idea as well?